src/server/scripts/ExilesReach/zone_exiles_reach.cpp

```diff
@@ -5920,6 +5925,886 @@ class quest_resizing_the_situation_horde : public quest_resizing_the_situation
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+private:
+    EventMap _events;
+    bool _endOfScene;
+};
+
+enum Torgok
+{
+    EVENT_CAST_SPIRIT_BOLT              = 1,
+    EVENT_CAST_SOUL_GRASP               = 2,
+
+    ITEM_TORGOKs_REAGENT_POUCH          = 176398,
+
+    QUEST_TORGOKs_REAGENT_POUCH_DROPPED = 59610,
+
+    SPELL_SPIRIT_BOLT_TORGOK            = 319294,
+    SPELL_SOUL_GRASP_TORGOK             = 319298
+};
+
+// 162817 - Torgok
+struct npc_torgok_q55879 : public ScriptedAI
+{
+    npc_torgok_q55879(Creature* creature) : ScriptedAI(creature) { }
+
+    void Reset() override
+    {
+        _events.Reset();
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        Talk(SAY_AGGRO, who);
+
+        _events.ScheduleEvent(EVENT_CAST_SPIRIT_BOLT, 4s);
+        _events.ScheduleEvent(EVENT_CAST_SOUL_GRASP, 14s);
+    }
+
+    void JustDied(Unit* killer) override
+    {
+        Talk(SAY_DEATH, killer);
+
+        for (auto const& [playerGuid, loot] : me->m_personalLoot)
+        {
+            if (Player* player = ObjectAccessor::GetPlayer(*me, playerGuid))
+            {
+                for (LootItem const& lootItem : loot->items)
+                {
+                    if (lootItem.itemid == ITEM_TORGOKs_REAGENT_POUCH)
+                    {
+                        player->SetRewardedQuest(QUEST_TORGOKs_REAGENT_POUCH_DROPPED);
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
+            return;
+
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_CAST_SPIRIT_BOLT:
+                    DoCastVictim(SPELL_SPIRIT_BOLT_TORGOK);
+                    _events.ScheduleEvent(EVENT_CAST_SPIRIT_BOLT, 6s);
+                    break;
+                case EVENT_CAST_SOUL_GRASP:
+                    DoCastAOE(SPELL_SOUL_GRASP_TORGOK);
+                    _events.ScheduleEvent(EVENT_CAST_SOUL_GRASP, 14s);
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+private:
+    EventMap _events;
+};
+
+enum HenryGarrickQuest55879
+{
+    PATH_PRISONER_TO_GROUND = 80000230
+};
+
+// 156799 - Henry Garrick
+struct npc_henry_q55879 : public ScriptedAI
+{
+    npc_henry_q55879(Creature* creature) : ScriptedAI(creature) { }
+
+    void InitializeAI() override
+    {
+        me->RemoveNpcFlag(UNIT_NPC_FLAG_QUESTGIVER);
+    }
+
+    void JustAppeared() override
+    {
+        me->RemoveAllAuras();
+        me->GetMotionMaster()->MovePath(PATH_PRISONER_TO_GROUND, false);
+    }
+};
+
+CreatureAI* HenryGarrickSelector(Creature* creature)
+{
+    if (creature->IsPrivateObject())
+        return new npc_henry_q55879(creature);
```

Comment:
i'd prefer to keep the `_private` postfix as we used to for all other private summons AIs so far